docs/federation.md

```diff
@@ -0,0 +1,126 @@
+* Only one AMKO can be a leader, all other AMKOs have to be followers. If there are two leaders at any point, federation will stop and the error will be written to the `AMKOCluster`'s status.
+
+### Disasters and Recovery
+Whenever a disaster occurs and it takes down the cluster where the leader AMKO is deployed, the kubernetes/openshift objects will not be synced to the Avi GSLB Leader Controller anymore. At this point, the user has to switch one of the follower AMKOs to leader and the syncing should just resume.
+
+Let's assume that there are 3 sites with one cluster in each of them:
+1. Cluster 1 in Site 1
+2. Cluster 2 in Site 2
+3. Cluster 3 in Site 3
+
+A disaster can occur either in the entire site or just for that cluster. Site failure would also mean that the kubernetes/openshift cluster along with the Avi Controller in that site are down. Whereas, a cluster failure would mean that only the kubernetes/openshift cluster is down.
+
+#### Site Failure Scenario
+
+ ![Alt text](images/amko_fed_switch.png?raw=true "AMKO site failure switch")
+
+The site where the leader AMKO was deployed and which hosted the Avi GSLB leader, fails. At this point, the user has to:
+1. Choose a new Avi GSLB leader controller out of the GSLB followers by following [these](https://avinetworks.com/docs/20.1/how-to-select-a-leader-site-manually/) steps.
+2. Choose a new follower AMKO to be the new leader, and perform the following steps on the cluster where this follower AMKO is deployed:
+  a. Edit the `GSLBConfig` object and change the leader IP address:
+
+        $ kubectl edit gslbconfig -n avi-system gc-1
+        // Set the field spec.gslbLeader.controllerIP to the new leader's IP address
+
+  b. Set the `isLeader` field in the `AMKOCluster` object to true on this cluster:
+
+    $ kubectl edit amkocluster amkocluster-federation -n avi-system
+
+This would reboot AMKO in leader mode and will resume syncing objects to the Avi GSLB leader controller.
+
+#### Cluster Failure Scenario
+
+ ![Alt text](images/amko_fed_cluster_failure.png?raw=true "AMKO cluster failure switch")
+
+The cluster where the leader AMKO was deployed, fails. Since the Avi GSLB leader is still active, the user only has to choose a new AMKO leader out of the followers. On that cluster, the `AMKOCluster` object must be edited and `spec.isLeader` field must be set to `true`:
+```
+$ kubectl edit amkocluster amkocluster-federation -n avi-system
+```
+
+#### Old leader AMKO boots up
+Note that at any point in time, there cannot be more than one leader AMKOs. If a leader AMKO finds that there's another leader in its member list, it won't federate. And, this AMKO won't sync any objects to the Avi GSLB leader controller. While the traffic will still continue to work, the kubernetes/openshift objects won't be synced to Avi GSLB leader any more. The user must intervene and set one of these AMKOs to follower by setting `spec.isLeader` field to `false` in the `AMKOCluster` object:
```

Comment:
This section can be just changed to:

At a given point in time, the architecture only allows a single AMKO `leader`. Conflicts leading from more than 1 leader must be resolved by the admin manually. This does not have any traffic impact on the existing GSLBService objects.